### ISSUE TITLE
Wrong ordering in CollectionEngine

### ISSUE THREAD

**Issue by andycaramba** [yajra/laravel-datatables#413]
Hi.

You have made changes in [425f65bf260d56ebe23d5cba3256113e12018e78](https://github.com/yajra/laravel-datatables/blob/425f65bf260d56ebe23d5cba3256113e12018e78/src/Engines/CollectionEngine.php#L108) and began to use `getColumnNameByIndex` instead of `getColumnName`. But `getColumnNameByIndex` does not return the correct column name, if the column order is different in the collection and datatable. When i've change back to `getColumnName` ordering works right.

Sorry for my English.

**Comment by yajra** [comment]
@andycaramba, I just actually renamed that function to appropriate one but the content did not change much to break it as far as I know? Can you please paste some snippets that replicates your issue? Demo seems to work fine.

**Comment by andycaramba** [comment]
I have this code in my controller
```php
public function getData(LogRepository $repository, $file)
{
    $repository->setFile(base64_decode($file));

    $logs = $repository->all(); // <= returns Illuminate\Support\Collection

    return Datatables::of($logs)->make(true);
}
```
and this js snippet in my view template
```javascript
$('#logs-table').DataTable({
    processing: true,
    serverSide: true,
    ajax: "{{ route('backend.logs.index.data', ['file' => base64_encode($current_file)]) }}",
    order: [[3, 'desc']],
    columns: [
        {data: 'user_id', name: 'user_id'},
        {data: 'module', name: 'module'},
        {data: 'level_name', name: 'level_name'},
        {data: 'datetime', name: 'datetime', type: 'date'},
        {data: 'message', name: 'message', 'class': 'long-text'}
    ]
});
```
Each element in my collection looks like
![image](https://cloud.githubusercontent.com/assets/229699/13356007/0418565e-dcc5-11e5-9aae-649d8a36f182.png)

I've put dd() in `CollectionEngine::ordering()` method and get next results

When I use `getColumnNameByIndex` with this code
![image](https://cloud.githubusercontent.com/assets/229699/13356303/60bac3c8-dcc6-11e5-83b8-83154cc1c834.png)
I have this dd output
![image](https://cloud.githubusercontent.com/assets/229699/13356315/71b0e392-dcc6-11e5-803d-0776025f556e.png)

But when I use `getColumnName`
![image](https://cloud.githubusercontent.com/assets/229699/13356412/eb69aa34-dcc6-11e5-98e6-db78c61549f0.png)
I have this dd output
![image](https://cloud.githubusercontent.com/assets/229699/13356457/29d7583e-dcc7-11e5-86c5-989baff32f00.png)

**Comment by yajra** [comment]
Can you please test `dev-develop` branch if it fixes the issue?

**Comment by yajra** [comment]
Patch was released on v6.7.2. Thanks for the hint and sorry for the inconvenience. Regards!
